scipy/stats/_kde.py

```diff
@@ -596,30 +596,27 @@ def logpdf(self, x):
                 points = reshape(points, (self.d, 1))
                 m = 1
             else:
-                msg = "points have dimension %s, dataset has dimension %s" % (d,
-                    self.d)
+                msg = (f"points have dimension {d}, "
+                       f"dataset has dimension {self.d}")
                 raise ValueError(msg)
 
-        if m >= self.n:
-            # there are more points than data, so loop over data
-            energy = np.empty((self.n, m), dtype=float)
-            for i in range(self.n):
-                diff = self.dataset[:, i, newaxis] - points
-                tdiff = dot(self.inv_cov, diff)
-                energy[i] = sum(diff*tdiff, axis=0)
-            log_to_sum = 2.0 * np.log(self.weights) - self.log_det - energy.T
-            result = logsumexp(0.5 * log_to_sum, axis=1)
+        output_dtype = np.common_type(self.covariance, points)
+        itemsize = np.dtype(output_dtype).itemsize
+        if itemsize == 4:
```

Comment:
Are these already tested?

Reply:
`test_kde_output_dtype` didn't address these lines because this test check the behavior of not `logpdf` (or `evaluate`) but `stats.gaussian_kde`.

Reply:
This `itemsize` block looks like a very weird check, almost certainly not the right thing to do. What is it trying to achieve?

Reply:
Thank you for the review. I copied this block from `evaluate(pdf)` so that this function is consistent with `pdf`. `logpdf` and `pdf` have the same functionality except for output values. In this sense, I thought the implementations of `logpdf` and `pdf` also should be similar to each other. I understand this block looks meaningless, and I agree to remove this block. However, to decide whether to remove it, we may have to check how the https://github.com/scipy/scipy/pull/10824 (I copied from this PR) was processed.

Reply:
Makes sense, I looked at the history and it's okay to keep. I'll push a refactor to avoid having two copies of that code block floating around.

Reply:
Thank you!